### PR TITLE
feat(core): surface nonPrintableRatio to catch CMap-garbage PDFs

### PR DESCRIPTION
**Pull request by yamadashy**
## Summary

- Add `nonPrintableRatio` (0..1, 3dp) on `PageResult` and `PageOverview` to expose a third silent-failure class: PDFs whose fonts lack a ToUnicode CMap (Hebrew, older CJK, missing-embed PDFs) where pdf.js falls back to raw glyph indices. The resulting `text` is full of NUL / C0 / C1 controls but `textCoverage` reads 100% — the existing low-coverage heuristic never fires.
- Markdown overview shows a `NonPrint` column only when at least one page is non-zero; the per-page header inlines `nonPrint: N%` with the same conditional. XML always carries `nonPrintableRatio` as an attribute for JSON parity.
- Skill docs (`SKILL.md`, `references/structured-output.md`) updated with the new field and the `>= 0.05` heuristic.

Real-world verification on `udhr-hebrew.pdf` (Unicode.org sample):

| Page | Chars | Coverage | **NonPrint** |
| ---: | ---: | ---: | ---: |
| 1 | 2760 | 100% | **62%** |
| 2 | 2314 | 100% | **59%** |
| 3 | 2487 | 100% | **61%** |
| 4 | 990 | 99% | **60%** |

Native-text samples (attention paper, Apple 10-K, EU AI Act, PDF spec, Korean / Chinese UDHR) all stay at 0.

## Design

Consulted Codex on category selection and naming. Outcome:
- **Count**: NUL, C0 except `\t\n\r`, DEL, C1, unpaired surrogates, Unicode noncharacters (`FDD0..FDEF`, plane-end `FFFE/FFFF`).
- **Don't count**: Private Use Area (math / dingbats use PUA legitimately), Cf format controls (ZWJ / variation selectors / bidi marks), combining marks. Mixing these in would dilute the "raw bytes" signal.
- **Computed on**: the post-NFKC `text` we actually return — matches what agents read.
- **Always on**: it's a single string walk, cheap enough not to need a flag.
- **Threshold (skill doc only)**: `>= 0.05` suspicious / `>= 0.3` pathological / `< 0.01` normal. pdfvision exposes the ratio; the agent decides.

A second signal (`renderContentRatio`, for catching blank-PNG renders from CMap / JPX) is planned as a follow-up PR — it touches renderer + OCR + cache so deserves its own diff.

Cache version: `structured-v8` → `structured-v9`.

## Test plan

- [x] `npm run lint` clean
- [x] `npm test` 195/195 passing (added 11 new unit tests covering edge cases: empty string, ASCII, tab/newline allowed, NUL-only garbage, mixed C0/C1, PUA excluded, format-controls excluded, noncharacters, lone surrogate, surrogate-pair single code-point counting, 3dp rounding)
- [x] Manual: Hebrew UDHR overview now flags 60% non-printable
- [x] Manual: clean PDFs (attention paper, Apple 10-K, Korean / Chinese UDHR) report 0
- [x] Manual: markdown overview column appears only when non-zero

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/core/nonPrintable.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * matching `textCoverage`'s rounding so the two signals compare like-for-like.
+ */
+export function nonPrintableRatio(text: string): number {
+  if (text.length === 0) return 0;
+  let total = 0;
+  let bad = 0;
+  for (const ch of text) {
+    total++;
+    if (isNonPrintableCodePoint(ch.codePointAt(0) as number)) bad++;
+  }
```
[the remark this answers is not in the document]
Comment (yamadashy) [reply src/core/nonPrintable.ts]: Addressed in `38aff9b` — switched to a manual `codePointAt` index walk so we stop allocating a fresh string per code point. 🤖
